server/src/main/java/org/opensearch/index/engine/IngestionEngine.java

```diff
@@ -0,0 +1,1016 @@
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.common.util.concurrent.AbstractRunnable;
+import org.opensearch.common.util.concurrent.ReleasableLock;
+import org.opensearch.common.util.io.IOUtils;
+import org.opensearch.core.common.unit.ByteSizeValue;
+import org.opensearch.core.index.shard.ShardId;
+import org.opensearch.index.IndexSettings;
+import org.opensearch.index.IngestionConsumerFactory;
+import org.opensearch.index.IngestionShardConsumer;
+import org.opensearch.index.IngestionShardPointer;
+import org.opensearch.index.mapper.DocumentMapperForType;
+import org.opensearch.index.mapper.IdFieldMapper;
+import org.opensearch.index.mapper.ParseContext;
+import org.opensearch.index.merge.MergeStats;
+import org.opensearch.index.merge.OnGoingMerge;
+import org.opensearch.index.seqno.SeqNoStats;
+import org.opensearch.index.shard.OpenSearchMergePolicy;
+import org.opensearch.index.translog.NoOpTranslogManager;
+import org.opensearch.index.translog.Translog;
+import org.opensearch.index.translog.TranslogCorruptedException;
+import org.opensearch.index.translog.TranslogManager;
+import org.opensearch.index.translog.TranslogStats;
+import org.opensearch.indices.ingest.DefaultStreamPoller;
+import org.opensearch.indices.ingest.StreamPoller;
+import org.opensearch.search.suggest.completion.CompletionStats;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.UnaryOperator;
+
+/**
+ * IngestionEngine is an engine that ingests data from a stream source.
+ */
+public class IngestionEngine extends Engine {
+
+    private volatile SegmentInfos lastCommittedSegmentInfos;
+    private final CompletionStatsCache completionStatsCache;
+    private final IndexWriter indexWriter;
+    private final OpenSearchReaderManager internalReaderManager;
+    private final ExternalReaderManager externalReaderManager;
+    private final Lock flushLock = new ReentrantLock();
+    private final ReentrantLock optimizeLock = new ReentrantLock();
+    private final OpenSearchConcurrentMergeScheduler mergeScheduler;
+    private final AtomicBoolean shouldPeriodicallyFlushAfterBigMerge = new AtomicBoolean(false);
+    private final TranslogManager translogManager;
+    private final DocumentMapperForType documentMapperForType;
+    private final IngestionConsumerFactory ingestionConsumerFactory;
+    protected StreamPoller streamPoller;
```

Comment:
It looks like `streamPoller` is assigned in the constructor and never accessed outside this class. Why is it not `private final`?